### PR TITLE
Add scripted_metric agg context to unsigned_long

### LINKED ISSUE #64347: Script Error when using unsigned long in a scripted_metric
Executing a `scripted_metric` on an `unsigned_long` field fails, while the same works for long.

Minimal repro case:

```
PUT ul_test
{
  "mappings": {
    "properties": {
      "a": {
        "type": "unsigned_long"
      }
    }
  }
}

POST ul_test/_doc/1
{
  "a": 42
}

PUT l_test
{
  "mappings": {
    "properties": {
      "a": {
        "type": "long"
      }
    }
  }
}

POST l_test/_doc/1
{
  "a": 42
}

GET l_test/_search
{
  "aggs": {
    "sqrt": {
      "scripted_metric": {
        "init_script": "state.sqrd = []",
        "map_script": "state.sqrd.add(doc.a.value * doc.a.value)",
        "combine_script": "state.sqrd",
        "reduce_script": "def sum = 0.0; for(l in states){ for(a in l) { sum += a}} return sum"
      }
    }
  }
}

GET ul_test/_search
{
  "aggs": {
    "sqrt": {
      "scripted_metric": {
        "init_script": "state.sqrd = []",
        "map_script": "state.sqrd.add(doc.a.value * doc.a.value)",
        "combine_script": "state.sqrd",
        "reduce_script": "def sum = 0.0; for(l in states){ for(a in l) { sum += a}} return sum"
      }
    }
  }
}
```

error:
```
{
  "error" : {
    "root_cause" : [
      {
        "type" : "script_exception",
        "reason" : "runtime error",
        "script_stack" : [
          "state.sqrd.add(doc.a.value * doc.a.value)",
          "                    ^---- HERE"
        ],
        "script" : "state.sqrd.add(doc.a.value * doc.a.value)",
        "lang" : "painless",
        "position" : {
          "offset" : 20,
          "start" : 0,
          "end" : 41
        }
      }
    ],
```

### PR DESCRIPTION
**Pull request by mayya-sharipova**
Also enhance documentation to provide more examples
how unsigned_long field should be used in scripts

Relates to #60050
Closes #64347

### CODE REVIEW

**Comment by mayya-sharipova** [comment]
@elasticmachine run elasticsearch-ci/2

**Comment by mayya-sharipova** [comment]
Stalled because of https://github.com/elastic/elasticsearch/issues/64361, until we make a decision how to handle `unsigned_long` in scripts.

**Comment by jimczi** [comment]
Is it still needed @mayya-sharipova ?

**Comment by mayya-sharipova** [comment]
@elasticmachine run elasticsearch-ci/part-2

**Comment by mayya-sharipova** [comment]
I've added changes based on @stu-elastic's PR #76519. @stu-elastic would appreciate a review from you.

File: x-pack/plugin/mapper-unsigned-long/src/yamlRestTest/resources/rest-api-spec/test/50_script_values.yml [hunk 1]
```diff
@@ -196,3 +199,36 @@ setup:
...
+          query:
+            term:
+              ul: 18446744073709551615
+          aggs:
+            sqrt:
+              scripted_metric:
+                init_script: |
+                  state.sqrd = []
+                map_script: |
+                  double v = field('ul').asBigInteger(BigInteger.ZERO).doubleValue();
```
Comment (stu-elastic) [inline x-pack/plugin/mapper-unsigned-long/src/yamlRestTest/resources/rest-api-spec/test/50_script_values.yml:222]: This is added in 7.16

**Review by stu-elastic — APPROVED** [review]
lgtm once the version check is updated

File: x-pack/plugin/mapper-unsigned-long/src/yamlRestTest/resources/rest-api-spec/test/50_script_values.yml (hunk 1 above)
Comment (mayya-sharipova) [reply x-pack/plugin/mapper-unsigned-long/src/yamlRestTest/resources/rest-api-spec/test/50_script_values.yml:222]: @stu-elastic Thanks for your feedback. Indeed `field` API was added in 7.16, but `scripted_metric` agg context was added to `unsigned_long` only from 8.1 ( I should correct `8.0` to `8.1`).

**Review by nik9000 — APPROVED** [review]

**Pull request merged**